proto/expr.proto

```diff
@@ -155,10 +155,15 @@ message TableFunction {
     UNNEST = 2;
     REGEXP_MATCHES = 3;
     RANGE = 4;
+    // User defined table function
+    UDTF = 100;
   }
   Type function_type = 1;
   repeated expr.ExprNode args = 2;
-  data.DataType return_type = 3;
+  repeated data.DataType return_types = 3;
```

Comment:
PTAL @xxchan . so I think we will need to check the returning column number when planning `ProjectSet`.

Reply:
No, in PG, table functions only return one columns. It's row expression instead of multi column.

```sql
xxchan@/tmp:postgres> create function foo(x int, y int) returns table (a int, b int) as $$ SELECT x, y $$ language sql;
CREATE FUNCTION
Time: 0.005s
xxchan@/tmp:postgres> select foo(1,2);
+------------+
| foo        |
|------------|
| ('1', '2') |
+------------+
SELECT 1
Time: 0.003s
xxchan@/tmp:postgres> explain select foo(1,2);
+--------------------------------------------------+
| QUERY PLAN                                       |
|--------------------------------------------------|
| ProjectSet  (cost=0.00..5.27 rows=1000 width=32) |
|   ->  Result  (cost=0.00..0.01 rows=1 width=0)   |
+--------------------------------------------------+
EXPLAIN 2
xxchan@/tmp:postgres> select 1, foo(1,2);
+----------+------------+
| ?column? | foo        |
|----------+------------|
| 1        | ('1', '2') |
+----------+------------+
```

Reply:
The column names are stored

```sql
select * from foo(1,2);
+---+---+
| a | b |
|---+---|
| 1 | 2 |
+---+---+

select (foo(1,2)).*;
+---+---+
| a | b |
|---+---|
| 1 | 2 |
+---+---+
```

Reply:
https://risingwave-labs.slack.com/archives/C048A6P0CSJ/p1668063950734419

Reply:
So it seems that this change should be reverted.

Reply:
I've reverted it to a single type.
For single column return table, e.g. `return table (a int)`, should the `return_type` be `int` or `struct<a: int>`?